src/OpenColorIO/transforms/builtins/ACES.cpp

```diff
@@ -0,0 +1,296 @@
+     0.010000000000000, -2.721718645000000,
+     0.028000000000000, -2.521718645000000,
+     0.054000000000000, -2.321718645000000,
+     0.095000000000000, -2.121718645000000,
+     0.145000000000000, -1.921718645000000,
+     0.220000000000000, -1.721718645000000,
+     0.300000000000000, -1.521718645000000,
+     0.400000000000000, -1.321718645000000,
+     0.500000000000000, -1.121718645000000,
+     0.600000000000000, -0.926545676714876
+};
+
+void GenerateOps(OpRcPtrVec & ops)
+{
+    static constexpr double CDD_TO_CID[4 * 4]
+    {
+        0.75573,  0.22197,  0.02230,  0.,
+        0.05901,  0.96928, -0.02829,  0.,
+        0.16134,  0.07406,  0.76460,  0.,
+        0.,       0.,       0.,       1.
+    };
+
+    // Convert Channel Dependent Density values into Channel Independent Density values.
+    CreateMatrixOp(ops, &CDD_TO_CID[0], TRANSFORM_DIR_FORWARD);
+
+    auto GenerateLutValues = [](double in) -> float
+    {
+        double out = 0.;
+
+        if (in < nonuniform_LUT[0]) // Lower bound i.e. in < nonuniform_LUT[0, 0].
+        {
+            // Extrapolate to ease conversion to LUT1D.
+            const double slope = (nonuniform_LUT[3] - nonuniform_LUT[1])
+                                    / (nonuniform_LUT[2] - nonuniform_LUT[0]);
+
+            out = nonuniform_LUT[1] - slope * (nonuniform_LUT[0] - in);
+
+            if (out < -10.) out = -10.;
+        }
+        else if (in <= nonuniform_LUT[(lutSize-1) * 2])
+        {
+            out = Interpolate1D(lutSize, &nonuniform_LUT[0], in);
+        }
+        else // Upper bound i.e. in > nonuniform_LUT[lutSize-1, 0].
+        {
+            const double REF_PT
+                = (7120. - 1520.) / 8000. * (100. / 55.) - std::log10(0.18);
+
+            out = (100. / 55.) * in - REF_PT;
+
+            if (out > 4.8162678) out = 4.8162678;  // log10(HALF_MAX)
+        }
+
+        return float(out);
+    };
+
+    // Convert Channel Independent Density values to Relative Log Exposure values.
+    CreateHalfLut(ops, GenerateLutValues);
+
+    // Convert Relative Log Exposure values to Relative Exposure values.
+    CreateLogOp(ops, 10., TRANSFORM_DIR_INVERSE);
+
+    static constexpr double EXP_TO_ACES[4 * 4]
+    {
+        0.72286,  0.12630,  0.15084,  0.,
+        0.11923,  0.76418,  0.11659,  0.,
+        0.01427,  0.08213,  0.90359,  0.,
+        0.,       0.,       0.,       1.
+    };
+
+    // Convert Relative Exposure values to ACES values.
+    CreateMatrixOp(ops, &EXP_TO_ACES[0], TRANSFORM_DIR_FORWARD);            
+}
+
+}  // namespace ADX
+
+
+namespace ACES
+{
+
+// Register all the ACES related built-in tranforms.
+void RegisterAll(BuiltinTransformRegistryImpl & registry) noexcept
+{
+    {
+        auto ACES_AP0_to_CIE_XYZ_D65_BFD_Functor = [](OpRcPtrVec & ops)
+        {
+            MatrixOpData::MatrixArrayPtr matrix
+                = build_conversion_matrix(ACES_AP0::primaries, CIE_XYZ_D65::primaries, ADAPTATION_BRADFORD);
```

Comment:
Possible candidate for public API. Static `MatrixTransform` constructor from primaries and CAT.

Reply:
Good point. But it should be revisited with the OCIO math library in mind.

Reply:
Yes, That makes sense.